### ISSUE TITLE
Allow exporting the Guid type in C#

### ISSUE THREAD

**Issue by nongvantinh** [godotengine/godot-proposals#7875]
### Describe the project you are working on

I'm developing an MMO game.

### Describe the problem or limitation you are having in your project

I need to implement the Inventory system, as well as other systems such as the Equipment system and the Ability System. These systems are implemented as `resources`, and each of them has the same requirement: they all need an ID for identification and loading the correct modifier from the database. However, due to limitations on Godot's side, I cannot export the Guid type in C# directly to the editor.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add the ability for [Export] attribute to also support the Guid type

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

`[Export] public Guid Id;`

With this code, I can save the ID in the .res file for consistency.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I don't know.

### Is there a reason why this should be core and not an add-on in the asset library?

It needs to be core.

**Comment by Calinou** [comment]
GDScript doesn't have a GUID Variant type, so how could this be achieved? The inspector doesn't have a way to display a GUID type either, but I guess it could be displayed as a string.

**Comment by nongvantinh** [comment]
Yeah, I currently have two variables to work around this issue: one to display the Id as `string` and the other to represent it as a Guid type. This allows me to convert it to a string type for display in the editor and saving it to the *.res file as a string. However, it would be nice if the editor could handle this transparently.

**Comment by Daylily-Zeleen** [comment]
Why not like this?

```
public static class Utils
{
	public static Guid ToGuid(this string str)
	{
		return Guid.Parse(str);
	}

}

public class MyTest: Godot.Node
{
        [Export]
	public string Id
	{
		get => Guid.ToString();
		set => Guid = value.ToGuid();
	}

	public Guid Guid;
}
```

Nothing need to be changed in godot source.
